src/chainstate/burn/operations/leader_block_commit.rs

```diff
@@ -224,6 +211,72 @@ impl LeaderBlockCommitOp {
             return Err(op_error::ParseError);
         }
 
+        let mut commit_outs = vec![];
+        let mut pox_fee = None;
+        let mut burn_fee = None;
+
+        for (ix, output) in outputs.into_iter().enumerate() {
+            // only look at the first OUTPUTS_PER_COMMIT outputs
+            //   or until first _burn_ output
+            if ix >= OUTPUTS_PER_COMMIT {
+                break;
+            }
+            if output.address.is_burn() {
+                burn_fee.replace(output.amount);
+                break;
+            } else {
+                // all pox outputs must have the same fee
+                if let Some(pox_fee) = pox_fee {
+                    if output.amount != pox_fee {
+                        warn!("Invalid commit tx: different output amounts for different PoX reward addresses");
+                        return Err(op_error::ParseError);
+                    }
+                } else {
+                    pox_fee.replace(output.amount);
+                }
+                commit_outs.push(output.address);
+            }
+        }
```

Comment:
This loop is specifying a strict schema for the outputs, I think it should be documented in a SIP.

Reply:
Yes, definitely, will update SIP-005

Reply:
Wait, shouldn't this be documented in SIP-007?

Reply:
This is an update to the wire format for block commits, so it should probably be SIP-005.

Reply:
Added to SIP-007 in 6c82029e6